src/pages/home/components/home-ranking/HomeRanking.tsx

```diff
@@ -0,0 +1,33 @@
+import { RankingWithJWTResponse } from '@/types/ranking';
+import { getRankings, getRankingsWithJWT } from '@/apis/ranking.api'; // 함수 임포트
+import * as S from './HomeRanking.style';
+import Loader from '@/components/loader/Loader';
+import HomeRankingItem from './item/HomeRankingItem';
+import { useQuery } from '@tanstack/react-query';
+
+const HomeRanking = ({ isJWT }: { isJWT: boolean }) => {
+  const { data, error, isLoading } = useQuery<RankingWithJWTResponse, Error>({
+    queryKey: ['data', isJWT],
+    queryFn: isJWT ? getRankingsWithJWT : getRankings,
+  });
```

Comment:
```suggestion
  const { data, error, isLoading } = useQuery<RankingWithJWTResponse | RankingResponse, Error>({
    queryKey: ['data', isJWT],
    queryFn: isJWT ? getRankingsWithJWT : getRankings,
  });
```

isJWT 유무에 따라 queryFn이 달라지기 때문에, 받는 데이터 타입도 위처럼 2가지로 지정해줘야 할 것 같습니다!
그리고, 이 data를 props로 넘겨주는 `<HomeRankingItem>` 컴포넌트에서도 수정해야 할 것 같아요!